controllers/stats/EditorialReportComponentHandler.inc.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * @file controllers/stats/EditorialReportComponentHandler.inc.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2000-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class EditorialReportComponentHandler
+ * @ingroup classes_controllers_stats
+ *
+ * @brief A class to create and compile configuration date for a Statistics UI
+ * component.
+ */
+class EditorialReportComponentHandler extends PKPHandler {
+	/** @var string The URL to the editorial report API endpoint */
+	public $_apiUrl = '';
+
+	/** @var array List of editorial statistics */
+	public $_editorialItems = [];
+
+	/** @var array List of user statistics */
+	public $_userItems = [];
+
+	/** @var array Configuration for the columns to display in the table */
+	public $_tableColumns = [];
+
+	/** @var string Retrieve stats after this date */
+	public $_dateStart = '';
+
+	/** @var string Retrieve stats before this date */
+	public $_dateEnd = '';
+
+	/** @var array Quick options to provide for configuring the date range */
+	public $_dateRangeOptions = [];
+
+	/** @var array|null Configuration assoc array for available filters */
+	public $_filters = null;
+
+	/** @var array Localized strings to pass to the component */
+	public $_i18n = [];
+
+	/** @var array List of submission stages and their statistics */
+	public $_submissionsStage = [];
+
+	/** @var array Data to present on the pie chart */
+	public $_editorialChartData = [];
+
+	/**
+	 * Constructor
+	 *
+	 * @param $apiUrl string The URL to fetch stats from
+	 * @param $args array Optional arguments
+	 */
+	public function __construct($apiUrl, $args = array()) {
+		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_MANAGER);
+		AppLocale::requireComponents(LOCALE_COMPONENT_APP_MANAGER);
+		parent::__construct();
+
+		$this->_apiUrl = $apiUrl;
+		$this->init($args);
+	}
+
+	/**
+	 * Given an array with statistics extracts the required editorial information to display in the component
+	 * @param $rangedStatistics array An array with a subset of the results from a call to PKPStatsService::getSubmissionStatistics
+	 * @param $statistics array An array with the results from a call to PKPStatsService::getSubmissionStatistics
+	 */
+	public static function extractEditorialStatistics($rangedStatistics, $statistics) {
+		$editorialStatistics = [];
+		$percentageField = 1;
+		$averageField = 2;
+		foreach ([
+			'SUBMISSION_RECEIVED' => [__('manager.statistics.editorial.submissionsReceived')],
+			'SUBMISSION_ACCEPTED' => [__('manager.statistics.editorial.submissionsAccepted')],
+			'SUBMISSION_DECLINED_TOTAL' => [__('manager.statistics.editorial.submissionsDeclined')],
+			'SUBMISSION_DECLINED_INITIAL' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.deskReject')],
```

Comment:
I've used this to apply some visual indentation to the sub-items. I was thinking about updating the component to work as a `tree listview`, but dropped the idea.